test/build-model.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { Response } from "express";
+import { buildModel } from "../src/index";
+import { MODEL_NOT_FOUND_ERROR } from "../src/constants";
+import { SequelizeQueryParserRequestInterface } from "../src/interfaces";
+
+const db = require("../example/db");
+describe("Build model middleware unit test", () => {
+  test("It must throw an error if model param is not associate with an existing model and not call next", () => {
+    const middleware = buildModel(db);
+    const fakeNext = jest.fn();
+    expect(() =>
+      middleware({ params: { model: "pet" } } as any, {} as Response, fakeNext)
+    ).toThrow(MODEL_NOT_FOUND_ERROR);
+    expect(fakeNext.mock.calls).toHaveLength(0);
+  });
+
+  test("It must throw an error if modelName is not associate with an existing model and not call next", () => {
```

Comment:
As before, this is a concatenation of the describe string, so name is _must throw..._ instead of _It must throw..._. Extend this to al test cases (AKA: _it(' '...)_ functions).